prover/crates/bin/circuit_prover/src/circuit_prover.rs

```diff
@@ -169,6 +182,45 @@ impl CircuitProver {
         let circuit_wrapper = prover_job.circuit_wrapper;
         let block_number = prover_job.block_number;
 
+        // snarkify begin
```

Comment:
nit: this whole block can be a lib, and reused by prover and compressor.

Reply:
yep, that was the plan